### PR TITLE
feat(trivy): Configure insecure image registries

### PR DESCRIPTION
**Pull request by xyoxo**
Resolves: https://github.com/aquasecurity/starboard/issues/545

### CODE REVIEW

File: pkg/starboard/config.go [hunk 1]
```diff
@@ -356,6 +357,17 @@ func (c ConfigData) GetTrivyServerURL() (string, error) {
 	return c.getRequiredProperty(keyTrivyServerURL)
 }
 
+func (c ConfigData) GetTrivyInsecureRegistries() []string {
```
Comment (danielpacak) [inline pkg/starboard/config.go:360]: I'd return a map[string]bool here so we don't duplicate the integration logic in plugin.go

Comment (danielpacak) [reply pkg/starboard/config.go:360]: Also we should have a unit test for new methods

File: pkg/plugin/trivy/plugin.go [hunk 2]
```diff
@@ -478,6 +495,21 @@ func (s *scanner) getPodSpecForClientServerMode(spec corev1.PodSpec, credentials
 			})
 		}
 
+		ref, err := name.ParseReference(container.Image)
```
Comment (danielpacak) [inline pkg/plugin/trivy/plugin.go:498]: The code for setting the TRIVY_INSECURE env is duplicated in two places: for Standalone and ClientServer mode. Can we instead refactor this logic to a private method and invoke it twice?

File: pkg/plugin/trivy/plugin.go [hunk 3]
```diff
@@ -478,6 +495,21 @@ func (s *scanner) getPodSpecForClientServerMode(spec corev1.PodSpec, credentials
...
+		if err != nil {
+			return corev1.PodSpec{}, nil, err
+		}
+		insecureRegistries := s.config.GetTrivyInsecureRegistries()
+		for _, insecureRegistry := range insecureRegistries {
+			if ref.Context().RegistryStr() == insecureRegistry {
+				env = append(env, corev1.EnvVar{
```
Comment (danielpacak) [inline pkg/plugin/trivy/plugin.go:505]: We should have a unit test scenario which asserts that the TRIVY_INSECURE env is appended.

**Review by danielpacak — CHANGES_REQUESTED** [review]
Good start @xyoxo See my comments before we can merge this PR

File: pkg/starboard/config.go (hunk 1 above)
Comment (xyoxo) [reply pkg/starboard/config.go]: updated to map[string]bool and added unit test.

File: pkg/plugin/trivy/plugin.go (hunk 2 above)
Comment (xyoxo) [reply pkg/plugin/trivy/plugin.go]: added a private method

File: pkg/plugin/trivy/plugin.go (hunk 3 above)
Comment (xyoxo) [reply pkg/plugin/trivy/plugin.go]: added unit test

**Review by danielpacak — APPROVED** [review]

**Pull request merged**